### PR TITLE
Override autoscale

### LINKED ISSUE #392: Add ability to modify price scale's price range
**Is your feature request related to a problem? Please describe.**

Yes, I want to be able to set the price scale's min and max value displayed.

**Describe the solution you'd like**

chart.setOption({ priceScale: { min: <value>, max: <value> } })

### PR DESCRIPTION
**Pull request by eugene-korobko**
**Type of PR:** enhancement

**PR checklist:**

- [x] Addresses an existing issue: fixes #392
- [x] Includes tests
- [x] Documentation update

Added ability to override autoscale result for series

### CODE REVIEW

File: docs/series-basics.md [hunk 1]
```diff
@@ -118,6 +118,62 @@ chart.applyOptions({
...
+            priceRange: {
+                minValue: 0,
+                maxValue: 100,
+            },
+        };
+    },
+});
+```
+
+So, you could just add a function that returns object with `priceRange` field to make chart thinking the series always has values from 0 to 100. These values are in prices.
```
Comment (ezhukovskiy) [inline docs/series-basics.md:139]:
```suggestion
So, you can just add a function that returns an object with `priceRange` field, if you want to override the original series' price range. For example, you can set the range from 0 to 100. These values are in prices.
```

File: docs/series-basics.md [hunk 2]
```diff
@@ -118,6 +118,62 @@ chart.applyOptions({
...
+
+You could also provide additional margins in pixels. Please be careful and never mix prices and pixels.
```
Comment (ezhukovskiy) [inline docs/series-basics.md:141]:
```suggestion
You can also provide additional margins in pixels. Please be careful and never mix prices and pixels.
```

File: docs/series-basics.md [hunk 3]
```diff
@@ -118,6 +118,62 @@ chart.applyOptions({
...
+            margins: {
+                above: 10,
+                below: 10,
+            },
+        };
+    },
+});
+```
+
+Actually, `autoscaleInfoProvider` function accepts a parameter with original implementation, so you can call default and adjust the result:
```
Comment (ezhukovskiy) [inline docs/series-basics.md:160]:
```suggestion
Actually, `autoscaleInfoProvider` function has an argument `original` which is the default implementation, so you can call it and adjust the result:
```

File: docs/series-basics.md [hunk 4]
```diff
@@ -118,6 +118,62 @@ chart.applyOptions({
...
+        if (res.priceRange !== null) {
+            res.priceRange.minValue -= 10;
+            res.priceRange.maxValue += 10;
+        }
+        return res;
+    },
+});
+```
+
+Please pay your attention that both `priceRange` and `margins` could be `null` in the default result.
```
Comment (ezhukovskiy) [inline docs/series-basics.md:175]:
```suggestion
Note that both `priceRange` and `margins` could be `null` in the default result.
```

File: src/model/price-scale.ts [hunk 5]
```diff
@@ -879,7 +879,7 @@ export class PriceScale {
 				}
 
 				if (autoScaleInfo !== null) {
-					const margins = autoScaleInfo.margins();
+					const margins = autoScaleInfo.margins() || null;
```
Comment (timocov) [inline src/model/price-scale.ts:882]: When this helps us?

Comment (eugene-korobko) [reply src/model/price-scale.ts:882]: If user overrides the autoscaleInfoProvider in options and the custom function does not return margins

Comment (timocov) [reply src/model/price-scale.ts:882]: So why we don't add `|| null` in AutoscaleInfoImpl?

Comment (Rostovsoft) [reply src/model/price-scale.ts]: Ok, we had a typing issue around it, now fixes

Comment (eugene-korobko) [reply src/model/price-scale.ts]: We had a typing issue around here, now fixed

**Pull request merged**